packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2507,6 +2534,7 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
       return new _CleanSemanticsFragment(renderObjectOwner: this);
     }
     List<_SemanticsFragment> children;
+    bool dropSemanticsOfLeftSiblings = (this is RenderBlockSemantics);
```

Comment:
Yeah, we don't want to do it this way, because it would prevent you from applying the same logic to e.g. RenderSector or RenderSliver subclasses. This should be something that any class can do. Maybe something you call on the SemanticsNode in your annotator?